back/src/domains/convention/use-cases/broadcast/BroadcastToPoleEmploiOnConventionUpdates.ts

```diff
@@ -25,6 +28,24 @@ const conventionObjectiveToObjectifDeImmersion: Record<
   "Initier une démarche de recrutement": 3,
 };
 
+const getLinkedAgencies = async (
+  agencyRepository: AgencyRepository,
+  convention: ConventionDto,
+): Promise<{ agency: AgencyDto; refersToAgency: AgencyDto | null }> => {
+  const agency = await agencyRepository.getById(convention.agencyId);
+  if (!agency) throw errors.agency.notFound({ agencyId: convention.agencyId });
+
+  let refersToAgency = null;
+  if (agency.refersToAgencyId) {
+    refersToAgency = await agencyRepository.getById(agency.refersToAgencyId);
+
+    if (refersToAgency === undefined)
+      throw errors.agency.notFound({ agencyId: agency.refersToAgencyId });
+  }
+
+  return { agency, refersToAgency };
+};
```

Comment:
```suggestion
const getLinkedAgencies = async (
  agencyRepository: AgencyRepository,
  convention: ConventionDto,
): Promise<{ agency: AgencyDto; refersToAgency: AgencyDto | null }> => {
  const agency = await agencyRepository.getById(convention.agencyId);
  if (!agency) throw errors.agency.notFound({ agencyId: convention.agencyId });

  if (!agency.refersToAgencyId) return { agency, refersToAgency:null }
  
  const  refersToAgency = await agencyRepository.getById(agency.refersToAgencyId);

  if (!refersToAgency) throw errors.agency.notFound({ agencyId: agency.refersToAgencyId });
  
  return { agency, refersToAgency };
};
```

On évite au max les `let` dans le code sauf au niveau des tests auto pour gérer les `before()`.